backend/junior-3/testing.md

```diff
@@ -0,0 +1,41 @@
+# Testing for Junior-3
+
+## Questions
+
+* What is automated testing, and how does it differ from manual?
+  * What is the regression bug problem and how automated testing helps solving it?
+  * What stages of manual testing are omitted by using automated testing instead?
+* What is a unit-test?
+  * Why is it useful to write unit-tests in groups?
+* How do tests help novice programmers?
+* How do tests help in writing project documentation?
+* How do tests help with receiving feedback from the coding process?
+Why does the feedback matter?
+* How do tests help refactoring process?
+* What is TDD?
+  * Why this approach to coding exists?
+  * What are the advantages of TDD?
+  * What are the limitations of TDD?
+* What are test doubles: dummies, fakes, stubs, spies, mocks?
+* What are test fixtures?
+* What are edge cases? Why do we test them?
+* What time do you think you should spend on a one programming cycle: `writing tests -> coding changes -> refactoring`?
+* Why does the speed of running all tests matter?
+* How to make the reason of test failure obvious?
+* What are integration tests?  
+How do unit-tests differ from integration tests?  
```

Comment:
This string lacks bullet list marker. And in some strings below too

Reply:
The questions would be under the same bullet marker https://github.com/fullstack-development/developers-roadmap/pull/130#discussion_r357901269